src/js/MainController.js

```diff
@@ -140,7 +140,9 @@ goog.require('ga_storage_service');
               if (!$window.olcs) {
                 // lazy load cesium library
                 $.getScript(ol3CesiumLibUrl, function() {
-                  $scope.ol3d = loadCesiumViewer($scope.map, active);
+                  $scope.$applyAsync(function() {
```

Comment:
$applyAsync triggers a $digest cycle if no digest cycle is currently running otherwise it use the current digest cycle.  That avoid a digest cycle if the user has triggered one by his action.